secret.go

```diff
@@ -57,8 +57,26 @@ const (
 //   ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*()-
 //
 // RandomPassword does not support Handler (user-invoked password rotation),
-// it only supports rotation by Secrets Manager.
-type RandomPassword struct{}
+// it only supports rotation by Secrets Manager. The password generated by
+// RandomPassword may be configured by providing RandomPasswordOptions on
+// initialization.
+type RandomPassword struct {
+	Opts *RandomPasswordOptions
+}
+
+// RandomPasswordOptions is used to configure a RandomPassword
+type RandomPasswordOptions struct {
```

Comment:
I think these options can/should be fields in `RandomPassword`. Option structs are usually used as params to func() calls, or parts of struct/objs that have a lot of fields and opts are just one small part of the exported fields. 

Example: https://golang.org/pkg/net/http/#Server some of the fields in that are the options; they configure the object. Similarly, these fields configure the `RandomPassword`.

Reply:
I think separating configurable options from the other components in a separate defined struct is cleaner, especially for larger more complex structs, but I don't feel super strongly about it/don't have a good enough reason so I made the changes.

Reply:
oh, I remember why I like it --- makes it super easy to pass in YAML configs